Makefile

```diff
@@ -1,67 +1,75 @@
+EXTAPI_CLIENT_ROOT = $(PROJECT_ROOT)/external
+EXTAPI_OPENAPI_DOC = $(PROJECT_ROOT)/swagger-external-$(ENGINE_REF).yaml
 
-.PHONY: build
-build: goversion ## Build all binaries
-ifeq (${VERBOSE}, 1)
-	go env
-endif
+# --- anchore enterprise references
+# ANCHORE_ENTERPRISE_REF = ...
 
-	@mkdir -p ${BUILD_DIR}
-	go build ${GOARGS} -tags "${GOTAGS}" -ldflags "${LDFLAGS}" -o ${BUILD_DIR}/ ./cmd/...
-
-.PHONY: build-release
-build-release: ## Build all binaries without debug information
-	@${MAKE} LDFLAGS="-w ${LDFLAGS}" GOARGS="${GOARGS} -trimpath" BUILD_DIR="${BUILD_DIR}/release" build
-
-.PHONY: build-debug
-build-debug: ## Build all binaries with remote debugging capabilities
-	@${MAKE} GOARGS="${GOARGS} -gcflags \"all=-N -l\"" BUILD_DIR="${BUILD_DIR}/debug" build
-
-.PHONY: goversion
-goversion:
-ifneq (${IGNORE_GOLANG_VERSION_REQ}, 1)
-	@printf "${GOLANG_VERSION}\n$$(go version | awk '{sub(/^go/, "", $$3);print $$3}')" | sort -t '.' -k 1,1 -k 2,2 -k 3,3 -g | head -1 | grep -q -E "^${GOLANG_VERSION}$$" || (printf "Required Go version is ${GOLANG_VERSION}\nInstalled: `go version`" && exit 1)
-endif
 
 define generate_openapi_client
+	# remove previous API clients
 	@ if [[ "$$OSTYPE" == "linux-gnu" ]]; then sudo rm -rf ${3}; else rm -rf ${3}; fi
-	docker run --rm -v $${PWD}:/local openapitools/openapi-generator-cli:${OPENAPI_GENERATOR_VERSION} generate \
-	--additional-properties packageName=${2} \
-	--additional-properties withGoCodegenComment=true \
-	-i /local/${1} \
-	-g go \
-	-o /local/${3}
+
+	# generate the API client
+	docker run \
+		--rm \
+		-v $${PWD}:/local \
+		openapitools/openapi-generator-cli:${OPENAPI_GENERATOR_VERSION} \
+			generate \
+				-c /local/generator.yaml \
+				--additional-properties packageName=${2} \
+				--additional-properties packageVersion=$(CLIENT_VERSION) \
+				--http-user-agent "anchore-client/$(CLIENT_VERSION)/go" \
+				-i /local/${1} \
+				-o /local/${3} \
+				-g go
+
+	# keep permissions the same as the user
 	@ if [[ "$$OSTYPE" == "linux-gnu" ]]; then sudo chown -R $(shell id -u):$(shell id -g) ${3}; fi
+
+	# remove unused files (go.mod is curated manually)
 	rm ${3}/{.travis.yml,git_push.sh,go.*}
 endef
 
-pkg/anchore/swagger.yaml:
-	curl https://raw.githubusercontent.com/anchore/anchore-engine/${ANCHORE_VERSION}/anchore_engine/services/apiext/swagger/swagger.yaml | tr '\n' '\r' | sed $$'s/- Images\r      - Vulnerabilities/- Images/g' | tr '\r' '\n' | sed '/- Image Content/d; /- Policy Evaluation/d; /- Queries/d' > pkg/anchore/swagger.yaml
+.PHONY :=
+.DEFAULT_GOAL :=
+update: clean generate-external-client ## pull all swagger definitions and generate client code
+
+.PHONY :=
+generate: generate-external-client ## generate all client code from all swagger documents
+
+$(EXTAPI_OPENAPI_DOC): ## pull the engine external API swagger document
+	mkdir -p $(PROJECT_ROOT)
+    # note: the existing upstream swagger document needs to be corrected, otherwise invalid code will be generated.
```

Comment:
is this outdented on purpose? or is it more of a spaces/tabs thing? - I don't mind either way, just making sure

Reply:
I think a mistake, I'll fix that 👍